fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/ResourceFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.models;
+
+/**
+ * Interface to a factory to locate or create new resources.
+ *
+ * @author whikloj
+ * @since 2019-09-23
+ */
+public interface ResourceFactory {
+
+    /**
+     * Find or create a new container.
+     *
+     * @param identifier The path or identifier for the resource.
+     * @param interactionModel The interaction model for the new/expected container.
+     * @return The container object.
+     */
+    public Container findOrInitContainer(final String identifier, final String interactionModel);
+
+    /**
+     * Find or create a new binary. Newly created binaries will have nothing backing them until they are populated by a
+     * service.
+     *
+     * @param identifier The path or identifier for the resource.
+     * @return The binary object.
+     */
+    public FedoraBinary findOrInitBinary(final String identifier);
```

Comment:
In order to figure out which binary class to initialize, you would need to also pass in `handling`

Reply:
This is the crux of the discussion I and @dbernstein were having, does this factory return the fully initialized and populated object?

Reply:
I don't think it needs to have all the incoming data from the client set at this point, and I take back this suggestion. I'll just copy my response from slack for the sake of having it here:

> Looking at the implementation of FedoraBinary, I’m now thinking it’d be fine not to include the handling parameter. It doesn’t initialize the implementation of the binary until its interacted with. So it would become a UrlBinary when its external binary url gets set.

Reply:
+1